### PR TITLE
OS: 36 Best Practices

### PR DESCRIPTION
**Pull request by shayaanmiyy**
Jira Ticket URL: https://digitalpolygon.atlassian.net/browse/OS-36

### Code Review:

- Review and refactor existing Twig templates for readability and maintainability.

- Apply best practices such as DRY (Don't Repeat Yourself) and KISS (Keep It Simple, Stupid).

### Template Caching:

- Implement caching strategies to optimize Twig template performance.

- Measure the performance improvements using tools like Query Monitor.

### Error Handling:

- Add error handling to Twig templates to gracefully handle missing data or other issues.

- Test your templates by simulating common errors and verifying the error handling works correctly.

image

### CODE REVIEW

File: web/app/themes/dp-timber-theme/views/partial/advance.twig [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <div class="container">
 	<form class="mt-4 mb-5">
 		<p class="m-0 lead">Here we using macro to print the input</p>
+        {# Here I used DRY, Instead of creating New Input I used Atoms #}
```
Comment (niteshsyngenta11) [inline web/app/themes/dp-timber-theme/views/partial/advance.twig]: Fix indenation issues.

File: web/app/themes/dp-timber-theme/views/partial/best-practices.twig [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<div class="container p-3 mt-5 bg-light rounded">
+	{# Conditional rendering based on user existence #}
+	{% if username is defined %}
+		<p>Hello,
+			{{ username }}</p>
+	{% else %}
+		<p>User not found.</p>
+	{% endif %}
```
Comment (niteshsyngenta11) [inline web/app/themes/dp-timber-theme/views/partial/best-practices.twig]: Fix indentation issues.

File: web/app/themes/dp-timber-theme/views/partial/best-practices.twig [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	<hr>
+
+	{# Providing a default message for posts #}
+	{% set firstPost = posts | first %}
+	<p>{{ firstPost.title | default('No posts available') }}</p>
+
+	<hr>
+
+	{# Using elseif to handle different roles #}
+	{% if user.role == 'admin' %}
```
Comment (niteshsyngenta11) [inline web/app/themes/dp-timber-theme/views/partial/best-practices.twig]: Use === instead ==

File: web/app/themes/dp-timber-theme/views/partial/best-practices.twig [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<div class="container p-3 mt-5 bg-light rounded">
+	{# Conditional rendering based on user existence #}
+	{% if username is defined %}
+		<p>Hello,
+			{{ username }}</p>
```
Comment (niteshsyngenta11) [inline web/app/themes/dp-timber-theme/views/partial/best-practices.twig]: This can go in a single line.

**Review by niteshsyngenta11 — CHANGES_REQUESTED** [review]
Added some comments.

File: web/app/themes/dp-timber-theme/views/partial/advance.twig (hunk 1 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/views/partial/advance.twig]: Yes, sure

File: web/app/themes/dp-timber-theme/views/partial/best-practices.twig (hunk 4 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/views/partial/best-practices.twig]: Yes, I will do it

File: web/app/themes/dp-timber-theme/views/partial/best-practices.twig (hunk 2 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/views/partial/best-practices.twig]: Yes

File: web/app/themes/dp-timber-theme/views/partial/best-practices.twig (hunk 3 above)
Comment (shayaanmiyy) [reply web/app/themes/dp-timber-theme/views/partial/best-practices.twig]: Yes, I fixed it

**Review by niteshsyngenta11 — APPROVED** [review]
Looks good 👍

**Pull request merged**